elfpy/pricing_models/base.py

```diff
@@ -124,11 +120,51 @@ def calc_bond_reserves(
         # Only want to renormalize time for APR ("annual", so hard coded to 365)
         # Don't want to renormalize stretched time
         annualized_time = time.norm_days(time_remaining.days, 365)
-        bond_reserves = (market_state.share_reserves / 2) * (
+        # y = z/2 * (mu * (1 + rt)**(1/tau) - c)
+        return (market_state.share_reserves / 2) * (
             market_state.init_share_price * (1 + target_apr * annualized_time) ** (1 / time_remaining.stretched_time)
             - market_state.share_price
-        )  # y = z/2 * (mu * (1 + rt)**(1/tau) - c)
-        return bond_reserves
+        )
+
+    def calc_bond_reserves(
+        self,
+        target_apr: float,
+        time_remaining: time.StretchedTime,
+        market_state: hyperdrive_market.MarketState,
+    ) -> float:
+        """Returns the assumed bond (i.e. token asset) reserve amounts given
+        the share (i.e. base asset) reserves and APR
+
+        Parameters
+        ----------
+        target_apr : float
+            Target fixed APR in decimal units (for example, 5% APR would be 0.05)
+        time_remaining : StretchedTime
+            Amount of time left until bond maturity
+        market_state : MarketState
+            MarketState object; the following attributes are used:
+                share_reserves : float
+                    Base asset reserves in the pool
+                init_share_price : float
+                    Original share price when the pool started
+                share_price : float
+                    Current share price
+
+        Returns
+        -------
+        float
+            The expected amount of bonds (token asset) in the pool, given the inputs
+
+        .. todo:: Test this function
+        """
+        # Only want to renormalize time for APR ("annual", so hard coded to 365)
+        annualized_time = time.norm_days(time_remaining.days, 365)
+        # (1 + r * t) ** (1 / tau)
+        interest_factor = (1 + target_apr * annualized_time) ** (1 / time_remaining.stretched_time)
+        # mu * z * (1 + apr * t) ** (1 / tau) - l
```

Comment:
where does this formula come from?
i'm lost as to what interest_factor is meant to accomplish
if annualized_time is a fraction of a year, then the amount of interest at the end of the period would be `(1 + r * t) ** t` but that's APY (with compounding) not APR. not sure why stretched time would factor in.